packages/app/src/cli/services/versions-list.ts

```diff
@@ -80,31 +81,42 @@ async function fetchAppVersions(
   }
 }
 
-async function getAppApiKey(token: string, options: VersionListOptions): Promise<string> {
-  if (options.apiKey) return options.apiKey
-  const envIdentifiers = getAppIdentifiers({app: options.app})
-  if (envIdentifiers.app) return envIdentifiers.app
-  if (isCurrentAppSchema(options.app.configuration)) return options.app.configuration.client_id
-  const partnersApp = await fetchOrCreateOrganizationApp(options.app, token)
-  return partnersApp.apiKey
-}
-
 interface VersionListOptions {
   app: AppInterface
   apiKey?: string
+  reset: boolean
+  commandConfig: Config
 }
 
 export default async function versionList(options: VersionListOptions) {
   const token = await ensureAuthenticatedPartners()
-  const apiKey = await getAppApiKey(token, options)
-  const {appVersions, totalResults, app} = await fetchAppVersions(token, apiKey)
+  const [partnersApp] = await fetchAppAndIdentifiers(options, token)
+  const {id: appId, organizationId, title, apiKey} = partnersApp
+
+  // remove early abort and revert the associated PR once unified deployment is rolled out 100%
+  const deploymentMode: DeploymentMode = partnersApp.betas?.unifiedAppDeployment ? 'unified' : 'legacy'
+  if (deploymentMode === 'legacy') {
+    renderError({
+      headline:
+        'The `app versions list` command is only available for apps that have upgraded to use simplified deployment.',
+      reference: [
+        {
+          link: {
+            label: 'Simplified extension deployment',
+            url: 'https://shopify.dev/docs/apps/deployment/simplified-deployment',
+          },
+        },
+      ],
+    })
+    throw new AbortSilentError()
+  }
```

Comment:
I will move this to the `context.ts` in a new method that could be named `ensureListContext`. Beasides, I will refactor the common code used by `ensureListContext` and `ensureReleaseContext` because I guess that once `unified deployment` is rolled out 100% the verification could be removed from both commands

Reply:
Hi @mattnoakes, yeah that's want I meant!, we could simply it a bit more if we move 
```ts
  const token = await ensureAuthenticatedPartners()
  const [partnersApp, envIdentifiers] = await fetchAppAndIdentifiers(options, token)
```

inside `checkDeploymentsBeta` and make the method to return the required values, but for me it's absolutely ok. 
Thanks again